hoomd/mpcd/RejectionVirtualParticleFiller.h

```diff
@@ -0,0 +1,152 @@
+
+#ifndef MPCD_REJECTION_FILLER_H_
+#define MPCD_REJECTION_FILLER_H_
+
+#ifdef NVCC
+#error This header cannot be compiled by nvcc
+#endif
+
+#include "VirtualParticleFiller.h"
+#include "hoomd/RandomNumbers.h"
+#include "hoomd/RNGIdentifiers.h"
+
+#include "hoomd/extern/pybind/include/pybind11/pybind11.h"
+
+namespace mpcd
+{
+
+//! Adds virtual particles to MPCD particle data for "sphere" geometry
+/*!
+ * <detailed description needed>
+*/
+
+template<class Geometry>
+class PYBIND11_EXPORT RejectionVirtualParticleFiller : public mpcd::VirtualParticleFiller
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param sysdata - MPCD system data
+         * \param density - number density of MPCD solvent
+         * \param type - particle fill type
+         * \param T - Temperature
+         * \param seed - seed for PRNG
+         * \param geom - confinement geometry
+         */
+        RejectionVirtualParticleFiller(std::shared_ptr<mpcd::SystemData> sysdata,
+                                       Scalar density,
+                                       unsigned int type,
+                                       std::shared_ptr<::Variant> T,
+                                       unsigned int seed,
+                                       std::shared_ptr<const Geometry> geom)
+        : mpcd::VirtualParticleFiller(sysdata, density, type, T, seed), m_geom(geom)
+        {
+        m_exec_conf->msg->notice(5) << "constructing MPCD RejectionVirtualParticleFiller" << std::endl;
+        }
+
+        //! Destructor
+        ~RejectionVirtualParticleFiller()
+        {
+        m_exec_conf->msg->notice(5) << "Destroying MPCD RejectionVirtualParticleFiller" << std::endl;
+        }
+
+        //! Get the streaming geometry
+        std::shared_ptr<const Geometry> getGeometry() const
+            {
+            return m_geom;
+            }
+
+        //! Set the streaming geometry
+        void setGeometry(std::shared_ptr<const Geometry> geom)
+            {
+            m_geom = geom;
+            }
+
+    protected:
+        std::shared_ptr<const Geometry> m_geom;
+
+        //! Fill the particles outside the confinement
+        void fill(unsigned int timestep);
+    };
+
+
+template<class Geometry>
+void RejectionVirtualParticleFiller<Geometry>::fill(unsigned int timestep)
+    {
+    // Number of particles that we need to draw (constant)
+    const BoxDim& box = m_pdata->getBox();
+    const Scalar3 lo = box.getLo();
+    const Scalar3 hi = box.getHi();
+    const unsigned int m_Lx = hi.x - lo.x;
+    const unsigned int m_Ly = hi.y - lo.y;
+    const unsigned int m_Lz = hi.z - lo.z;
+    const unsigned int m_NVirtMax = m_density*(m_Lx*m_Ly*m_Lz);
```

Comment:
I think `BoxDim` has a getVolume method:
```suggestion
    const Scalar3 lo = box.getLo();
    const Scalar3 hi = box.getHi();
    const unsigned int NVirtMax = round(m_density*box.getVolume());
```
In any case, do not down cast the bounds to integers, or you will get an error. Round to integer instead